folktables/folktables.py

```diff
@@ -141,7 +141,42 @@ def df_to_pandas(self, df, categories=None, dummies=False):
 
         return variables, target, group
 
-    @property 
+    def generate_categories(self, definition_df):
+        """
+        Generates a categories dictionary using the provided definition dataframe. Does not create a category mapping
+        for variable requiring the 2010 Public use microdata area code (PUMA) as these need an additional definition
+        file which are not unique without the state code.
+        """
+        categories = {}
+        for feature in self.features:
+            if 'PUMA' in feature:
+                continue
+
+            # extract definitions for this feature
+            coll_definition = definition_df[(definition_df[0] == 'VAL') & (definition_df[1] == feature)]
+
+            # extracts if the feature is numeric or categorical --> 'N' == numeric
+            coll_type = coll_definition.iloc[0][2]
+            if coll_type == 'N':
+                # do not add to categories
+                continue
+
+            # transform to numbers as definitions are strings.
```

Comment:
The following section could use better documentation. I find it hard to understand what's happening here. Is there a way to do it without using the `-99999999999999.0` trick?

Reply:
In addition, would you be able to add some tests for what's being implemented here? Thanks!